TicTacToe/Program.cs

```diff
@@ -7,415 +7,96 @@
-        }
-
+    {
          static void Main()
          {
-            StartGame: //restart from end of Main
-            
-            /* variable declaration:
-             * choice: selects where on board
-             * turn: toggles between x and o
-             * num: counts number of turns
-            */
-
-            int choice = 0, turn = 1, num = 1;
-            bool winFlag = false, playing = true;
-
-            DrawBoard(); //show players board
-
-            //game loop
-            while (playing == true)
+            while (true)
             {
-               
+                var winningPlayer = -1; //-1 null, 0 tie, 1 player 1 (X), 2 player 2 (O)
+                var totalTurns = 0;
+                var isGameOver = false;
+                var board = new Board();
 
-                while (winFlag == false)
+                while (!isGameOver)
                 {
-
+                    var currentPlayersTurn = ((totalTurns + 1) % 2) != 0 ? 1 : 2;  //ternary statements (cond ? if true : if false)/modulus operator (remainder from division)
+                    var isCurrentPlayersTurn = true;
+                    var validMove = false;
                     
-                    //player O turn
-                    if (turn == 1 && num < 10)
+                    while (isCurrentPlayersTurn)
                     {
+                        board.DrawBoard();
 
-                        Console.WriteLine("Player O's turn. Pick a number");
-                        choice = int.Parse(Console.ReadLine());
+                        Console.WriteLine($"Player {currentPlayersTurn}'s turn. Pick a number:"); //player 1 is X
 
-                        if (choice > 0 && choice < 10)
+                        var entry = Console.ReadLine();
+
+                        Console.WriteLine("You entered: " + entry);
+
+                        if (int.TryParse(entry, out var intChoice))
                         {
-                            if (pos[choice] != "X" && pos[choice] != "O")
+                            if (intChoice > 0 && intChoice < 10)
                             {
-                                pos[choice] = "O";
-                                winFlag = CheckWin();
-                                Console.Clear();
-                                DrawBoard();
-                                num++;
-
-                                if (winFlag == false)
-                                {
-                                    turn = 2;
-                                }
+                                var statusOfBoardAtChoice = board.Position.GetChoiceAtPosition(intChoice);
 
-                                else
+                                if (statusOfBoardAtChoice.ToLower() != "x" && statusOfBoardAtChoice.ToLower() != "o")
                                 {
-                                    playing = false;
+                                    board.Position.SetChoiceAtPosition(intChoice, currentPlayersTurn == 1);
```

Comment:
This actually broke the OOP contract. I should ask the board to do it for me rather than reach down another level and interact with one of it's dependencies.